public/apps/configuration/utils/tenant-utils.tsx

```diff
@@ -94,3 +109,60 @@ export function resolveTenantName(tenant: string, userName: string) {
     return tenant;
   }
 }
+
+function formatTenantName(tenantName: string): string {
+  if (tenantName === globalTenantName) return GLOBAL_USER_DICT.Label;
+  return tenantName;
+}
+
+export function transformRoleTenantPermissionData(
+  tenantPermissions: RoleTenantPermissionView[],
+  tenantList: Tenant[]
+): RoleTenantPermissionDetail[] {
+  return map(tenantPermissions, (tenantPermission: RoleTenantPermissionView) => {
+    const tenantNames: string[] = tenantList.map((t: Tenant) => t.tenant); // Global
+    let tenantItem = null;
+    if (
+      tenantPermission.tenant_patterns.length === 1 &&
+      tenantNames.includes(formatTenantName(tenantPermission.tenant_patterns[0]))
+    ) {
+      tenantItem = tenantList.filter((t) => {
+        return t.tenant === formatTenantName(tenantPermission.tenant_patterns[0]);
+      })[0];
+    }
+    return {
+      tenant_patterns: tenantPermission.tenant_patterns,
+      permissionType: tenantPermission.permissionType,
+      tenant: tenantItem ? tenantItem.tenant : RoleViewTenantInvalidText,
+      reserved: tenantItem ? tenantItem.reserved : false,
+      description: tenantItem ? tenantItem.description : RoleViewTenantInvalidText,
+      tenantValue: tenantItem ? tenantItem.tenantValue : RoleViewTenantInvalidText,
+    };
+  });
+}
+
+export function getTenantPermissionType(tenantPermissions: string[]) {
+  const readable = tenantPermissions.includes(TENANT_READ_PERMISSION);
+  const writable = tenantPermissions.includes(TENANT_WRITE_PERMISSION);
+  let permissionType = TenantPermissionType.None;
+  if (readable && writable) {
+    permissionType = TenantPermissionType.Full;
+  } else if (readable) {
+    permissionType = TenantPermissionType.Read;
+  } else if (writable) {
+    permissionType = TenantPermissionType.Write;
+  }
+  return permissionType;
+}
+
+export function transformRoleTenantPermissions(
+  roleTenantPermission: RoleTenantPermission[]
+): RoleTenantPermissionView[] {
+  return roleTenantPermission.map((tenantPermission) => {
+    const permissionType = getTenantPermissionType(tenantPermission.allowed_actions);
+    return {
+      tenant_patterns: tenantPermission.tenant_patterns,
+      permissionType,
+    };
```

Comment:
No need to declare `permissionType` so return keyword is not required. It can be shorten to 

~~~
(tenantPermission) => ({
      tenant_patterns: tenantPermission.tenant_patterns,
      permissionType: getTenantPermissionType(tenantPermission.allowed_actions)
});
~~~

Reply:
corrected.